experimental/dds/tree/src/EditLog.ts

```diff
@@ -529,7 +532,7 @@ export class EditLog<TChange = unknown> extends TypedEventEmitter<IEditLogEvents
 
 	private evictEdits(): void {
 		const minSequenceIndex =
-			this.sequenceNumberToIndex.getPairOrNextHigher(this._minSequenceNumber)?.[1] ??
+			this.sequenceNumberToIndex?.getPairOrNextHigher(this._minSequenceNumber)?.[1] ??
```

Comment:
What if somebody leaves `targetLength` at Infinity, but then then sets `evictionFrequency` to like 5? Won't this function get called every five edits and then fail here because `sequenceNumberToIndex` is undefined?

I think rather than using the `?.` (as much as I like terse syntax. The one on line 506 is fine.), we should be more clear about when `sequenceNumberToIndex` should be undefined and when it should not be. Perhaps `evictEdits` should fail entirely if we try to call it when `sequenceNumberToIndex` is undefined. I think my preference would be to do that and replace the conditional around the call `evictEdits` to be `sequenceNumberToIndex !== undefined`. Then, move the existing conditional (`this.numberOfSequencedEdits >= this.evictionFrequency`) inside of `evictEdits`, since `evictEdits` already has some conditionals in there that cause it to not actually do anything. Maybe we should call it `evictOldEdits` or `tryEvictEdits`, idk.

Reply:
Great point!